app/services/user-service.js

```diff
@@ -0,0 +1,158 @@
+import Ember from "ember";
+import ApiBaseService from "./api-base-service";
+import _ from "lodash";
+
+export default ApiBaseService.extend({
+  store: Ember.inject.service(),
+  session: Ember.inject.service(),
+  printerService: Ember.inject.service(),
+
+  adminAppRoles: ["Reviewer", "Supervisor"],
+  stockAppRoles: [
+    "Stock administrator",
+    "Stock fulfilment",
+    "Order administrator",
+    "Order fulfilment"
+  ],
+
+  canUpdateRole(userId) {
+    return (
+      this.get("session.currentUser.isAdministrator") &&
+      this.get("session.currentUser.canManageUserRoles") &&
+      +this.get("session.currentUser.id") !== +userId
+    );
+  },
+
+  getRoleId(roleName) {
+    return this.get("store")
+      .peekAll("role")
+      .find(role => role.get("name") === roleName)
+      .get("id");
+  },
+
+  getRoleExpiryDate(user, app) {
+    let roles = app == "admin" ? this.adminAppRoles : this.stockAppRoles;
+    let expiryDates = user
+      .get("activeUserRoles")
+      .filter(
+        row => _.includes(roles, row.get("role.name")) && !!row.get("expiresAt")
+      )
+      .map(row => row.get("expiresAt"));
+
+    return _.max(expiryDates);
+  },
+
+  getRoleAccessText(user, app) {
```

Comment:
could this be
```javascript
  getRoleAccessText(user, userType){
    .....
  }
```
where userType could be like a stock or admin user?
Using an app name could confuse